src/manifest/__tests__/period.test.ts

```diff
@@ -0,0 +1,436 @@
+
+    expect(period.parsingErrors).toHaveLength(0);
+    expect(period.adaptations.video).toHaveLength(2);
+
+    expect(adaptationSpy).toHaveBeenCalledTimes(2);
+    expect(adaptationSpy).toHaveBeenNthCalledWith(1, videoAda1, representationFilter);
+    expect(adaptationSpy).toHaveBeenNthCalledWith(2, videoAda2, representationFilter);
+    expect(representationFilter).not.toHaveBeenCalled();
+  });
+
+  it("should combine parsing errors from the Period and from the Adaptations", () => {
+    const adaptationSpy = jest.fn(arg => ({ ...arg, parsingErrors: [arg.id] }));
+    jest.mock("../adaptation", () => ({
+      default: adaptationSpy,
+      SUPPORTED_ADAPTATIONS_TYPE: ["audio", "video", "text", "image"],
+    }));
+
+    const Period = require("../period").default;
+    const videoAda1 = { type: "video", id: "54", representations: [{}] };
+    const videoAda2 = { type: "video", id: "55", representations: [{}] };
+    const fooAda1 = { type: "foo", id: "12", representations: [{}] };
+    const video = [videoAda1, videoAda2];
+    const foo = [fooAda1];
+    const args = { id: "12", adaptations: { video, foo }, start: 0 };
+    const period = new Period(args as any);
+
+    expect(period.parsingErrors).toHaveLength(3);
+    expect(period.parsingErrors).toContain("54");
+    expect(period.parsingErrors).toContain("55");
+  });
+
+  it("should set the given start", () => {
+    const adaptationSpy = jest.fn(arg => ({ ...arg, parsingErrors: [] }));
+    jest.mock("../adaptation", () => ({
+      default: adaptationSpy,
+      SUPPORTED_ADAPTATIONS_TYPE: ["audio", "video", "text", "image"],
+    }));
+
+    const Period = require("../period").default;
+    const videoAda1 = { type: "video", id: "54", representations: [{}] };
+    const videoAda2 = { type: "video", id: "55", representations: [{}] };
+    const video = [videoAda1, videoAda2];
+    const args = { id: "12", adaptations: { video }, start: 72 };
+    const period = new Period(args as any);
+    expect(period.start).toEqual(72);
+    expect(period.duration).toEqual(undefined);
+    expect(period.end).toEqual(undefined);
+  });
+
+  it("should set a given duration", () => {
+    const adaptationSpy = jest.fn(arg => ({ ...arg, parsingErrors: [] }));
+    jest.mock("../adaptation", () => ({
+      default: adaptationSpy,
+      SUPPORTED_ADAPTATIONS_TYPE: ["audio", "video", "text", "image"],
+    }));
+
+    const Period = require("../period").default;
+    const videoAda1 = { type: "video", id: "54", representations: [{}] };
+    const videoAda2 = { type: "video", id: "55", representations: [{}] };
+    const video = [videoAda1, videoAda2];
+    const args = { id: "12", adaptations: { video }, start: 0, duration: 12 };
+    const period = new Period(args as any);
+    expect(period.start).toEqual(0);
+    expect(period.duration).toEqual(12);
+    expect(period.end).toEqual(12);
+  });
+
+  it("should infer the end from the start and the duration", () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 294. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=439&issues=AWgpECqc_MPaBDn0IVTw&open=AWgpECqc_MPaBDn0IVTw)

[ISSUE_KEY]: AWgpECqc_MPaBDn0IVTw